wsl.md

```diff
@@ -0,0 +1,277 @@
+wsl.md
+
+<p align="center"><img src="/art/logo.svg" alt="Laravel Homestead Logo"></p>
+
+<p align="center"><span style="font-size: 24px; font-weight: bold;"> WSL - Windows Subsystem for Linux </span></p>
+
+
+- [Introduction](#introduction)
+- [Requirements](#requirements)
+- [Installation](#installation)
+  - [Installing Homestead](#installing-homestead)
+  - [Provision Homestead on WSL Distro](#provision-homestead-on-wsl-distro)
+  - [Installing Optional Features](#installing-optional-features)
+  - [Installing Databases](#installing-databases)
+  - [Configuring Nginx Sites](#configuring-nginx-sites)
+  - [Configuring Shared Folders](#configuring-shared-folders)
+    - [WSL network path for sharing folders](#wsl-network-path-for-sharing-folders)
+- [Aliases](#aliases)
+- [Applications and Features](#applications-and-features)
+  -  [Included Software](#included-software)
+  -  [Optional Software](#optional-software)
+
+
+## Introduction
+
+Welcome to the _"Initial Draft"_ documentation for using Homestead to provision WSL Distro for your local Laravel development.
+
+## Requirements
+
+**WSL Version**: WSL2
+
+Homestead for WSL is currently tested only with WSL version 2 (WSL2), though it can also work with WSL version 1.
+
+**WSL Distro**: Ubuntu 20.04 or Ubuntu 22.04
+
+Ensure that you have either Ubuntu 20.04 or Ubuntu 22.04 distributions installed. Homestead only supports these two distributions.
+
+To verify open a Powershell Terminal and use following command:
+
+```powershell
+> wsl -l -v
+  NAME            STATE           VERSION
+* Ubuntu          Running         2
+* Ubuntu-20.04    Running         2
+```
+The **'NAME'** column represents the distribution name, and the **'VERSION'** column identifies the WSL version.
+
+
+If you don't have any distribution installed, use the following command to list all available WSL distributions:
+```powershell
+> wsl -l --online
+The following is a list of valid distributions that can be installed.
+Install using 'wsl.exe --install <Distro>'.
+NAME                                   FRIENDLY NAME
+Ubuntu                                 Ubuntu
+Debian                                 Debian GNU/Linux
+kali-linux                             Kali Linux Rolling
+Ubuntu-18.04                           Ubuntu 18.04 LTS
+Ubuntu-20.04                           Ubuntu 20.04 LTS
+Ubuntu-22.04                           Ubuntu 22.04 LTS
+...
+```
+
+To install a distribution, use the following command:
+```powershell
+> wsl --install Ubuntu-22.04
```

Comment:
Should be wsl --install -d Ubuntu-22.04 isn't it ?

Reply:
@sdespont What does the -d option do? I can't even find it in the docs

[The docs](https://learn.microsoft.com/en-us/windows/wsl/basic-commands#install) say `wsl --install <Distribution Name>`

Reply:
The -d stands for "distribution", it permits to keep compatibility for older systems. Also, I am using Windows 23H2 but I need to specify the -d in the install command.
![image](https://github.com/laravel/homestead/assets/646632/d79a369e-40df-44ea-990e-b88908b9b202)